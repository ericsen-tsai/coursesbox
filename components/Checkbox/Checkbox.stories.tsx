import { ComponentStoryObj, ComponentMeta } from '@storybook/react'
import { expect } from '@storybook/jest'
import { screen, userEvent } from '@storybook/testing-library'

import Checkbox from './index'

export default {
  title: 'Controls/Checkbox',
  component: Checkbox,
} as ComponentMeta<typeof Checkbox>

export const PrimaryCheckbox: ComponentStoryObj<typeof Checkbox> = {
  play: async ({ args }) => {
    await userEvent.click(screen.getByLabelText('âœ“'))
    await expect(args.onChange).toHaveBeenCalled()
  },
}
