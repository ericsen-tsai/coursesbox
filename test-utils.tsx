import React, { FC, ReactElement, ReactNode } from 'react'

import { render, RenderOptions } from '@testing-library/react'

import { Themes } from './styles/themes'
import { ThemeProvider } from '@emotion/react'

const Wrapper = ({ children }: { children: ReactNode }) => {
  return <ThemeProvider theme={Themes.light}>{children}</ThemeProvider>
}

const customRender = (ui: ReactElement, options?: RenderOptions) =>
  render(ui, { wrapper: Wrapper, ...options })

export * from '@testing-library/react'

export { customRender as render }
