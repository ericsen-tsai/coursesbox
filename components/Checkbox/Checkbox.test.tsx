import { render, screen } from '@/test-utils'
import userEvent from '@testing-library/user-event'

import Checkbox from './index'

describe('Checkbox test cases', () => {
  it('Render check', () => {
    const onChange = jest.fn()
    jest.spyOn(Math, 'random').mockReturnValue(0.99999999999)
    const { asFragment } = render(<Checkbox onChange={onChange} />)

    expect(asFragment()).toMatchSnapshot()
  })

  it('Render check', async () => {
    const onChange = jest.fn()

    render(<Checkbox onChange={onChange} />)

    const element = screen.getByLabelText('âœ“')

    await userEvent.click(element)
    expect(onChange).toHaveBeenCalled()
  })
})
