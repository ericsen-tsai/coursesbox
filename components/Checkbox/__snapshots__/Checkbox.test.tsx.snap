// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox test cases Render check 1`] = `
<DocumentFragment>
  <label
    class="css-5y0pmi"
  >
    <input
      id="fffffffff5014"
      type="checkbox"
    />
    <label
      class="css-x787if"
      for="fffffffff5014"
    >
      âœ“
    </label>
  </label>
</DocumentFragment>
`;
